### ISSUE TITLE
Gitbash terminal 

### ISSUE THREAD

**Issue by rezadrian01** [microsoft/vscode#214538]
Type: <b>Performance Issue</b>

My gitbash terminal on vscode is very slowly, but when I open Gitbash directly through the application, it doesn't lag, and when i open other terminal like powershell its doesn't lag too

VS Code version: Code 1.90.0 (89de5a8d4d6205e5b11647eb6a74844ca23d2573, 2024-06-04T19:33:54.889Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 5800H with Radeon Graphics          (16 x 3194)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.41GB (5.07GB free)|
|Process Argv|--crash-reporter-id 702f5be6-4224-436f-9a70-a2cb56e59201|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	   112	 23720	code main
    0	   100	 10124	window [2] (Issue Reporter)
    0	    86	 13000	fileWatcher [1]
    0	   290	 23600	extensionHost [1]
    0	   104	  7844	     electron-nodejs (languageServer.js )
    0	   130	 19320	     "C:\Users\AHMAD REZA\AppData\Local\Programs\Microsoft VS Code\Code.exe" "c:\Users\AHMAD REZA\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\json-language-features\server\dist\node\jsonServerMain" --node-ipc --clientProcessId=23600
    0	    90	 20996	     "C:\Users\AHMAD REZA\AppData\Local\Programs\Microsoft VS Code\Code.exe" "c:\Users\AHMAD REZA\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\markdown-language-features\server\dist\node\workerMain" --node-ipc --clientProcessId=23600
    0	   113	 24616	     electron-nodejs (server.js )
    0	   120	 25012	     electron-nodejs (tsserver.js )
    0	    93	 25092	     electron-nodejs (serverMain.js )
    0	   121	 25160	     electron-nodejs (tailwindServer.js )
    0	   149	 25452	     electron-nodejs (tsserver.js )
    0	    90	 25308	       electron-nodejs (typingsInstaller.js typesMap.js )
    0	    92	 23632	ptyHost
    0	     5	 17908	     "C:\Program Files\Git\bin\bash.exe" --init-file "c:\Users\AHMAD REZA\AppData\Local\Programs\Microsoft VS Code\resources\app/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh"
    0	    11	 10024	       "C:\Program Files\Git\bin\..\usr\bin\bash.exe" --init-file "c:\Users\AHMAD REZA\AppData\Local\Programs\Microsoft VS Code\resources\app/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh"
    0	     7	 21532	     conpty-agent
    0	    29	 23812	   crashpad-handler
    0	   182	 23920	   gpu-process
    0	    51	 23988	   utility-network-service
    0	   360	 24072	window [1] (settings.json - NodeJS - The Complete Guide (MVC, REST APIs, GraphQL, Deno) - Visual Studio Code)
    0	   115	 24188	shared-process
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (settings.json - NodeJS - The Complete Guide (MVC, REST APIs, GraphQL, Deno) - Visual Studio Code)
|    Folder (NodeJS - The Complete Guide (MVC, REST APIs, GraphQL, Deno)): 461 files
|      File types: js(110) ejs(110) json(51) html(9) hbs(4) pug(4) css(3)
|                  gitignore(1)
|      Conf files: package.json(20);
```

</details>
<details><summary>Extensions (51)</summary>

Extension|Author (truncated)|Version
---|---|---
pug|ama|1.0.1
vscode-intelephense-client|bme|1.10.4
vscode-tailwindcss|bra|0.10.5
npm-intellisense|chr|1.4.5
postcss|css|1.0.9
dbclient-jdbc|cwe|1.3.5
vscode-database-client2|cwe|7.4.4
c-cpp-compile-run|dan|1.0.58
python-extensions-pack|dem|1.0.3
composer-php-vscode|DEV|1.47.15512
intelli-php-vscode|DEV|0.12.15062
phptools-vscode|DEV|1.47.15512
profiler-php-vscode|DEV|1.47.15512
javascript-ejs-support|Dig|1.3.3
prettier-vscode|esb|10.4.0
auto-close-tag|for|0.5.15
auto-rename-tag|for|0.1.10
code-runner|for|0.12.2
vscode-github-actions|git|0.26.3
vscode-pull-request-github|Git|0.88.1
mayukaithemevsc|Gul|3.2.4
search-node-modules|jas|1.3.0
language-julia|jul|1.79.2
mongodb-vscode|mon|1.6.0
debugpy|ms-|2024.6.0
python|ms-|2024.8.0
vscode-pylance|ms-|2024.6.1
jupyter|ms-|2024.5.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.18
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
cmake-tools|ms-|1.18.41
cpptools|ms-|1.20.5
cpptools-extension-pack|ms-|1.3.0
live-server|ms-|0.4.13
test-adapter-converter|ms-|0.1.9
LiveServer|rit|5.7.9
code-snapshot|rob|0.2.1
mongo-snippets-for-node-js|roe|1.3.12
vscode-pets|ton|1.26.0
cmake|twx|0.0.17
errorlens|use|3.18.0
highlight-matching-tag|vin|0.11.0
intellicode-api-usage-examples|Vis|0.2.8
vscodeintellicode|Vis|1.3.1
vscode-java-debug|vsc|0.57.0
vscode-java-pack|vsc|0.27.0
vscode-icons|vsc|12.8.0
quokka-vscode|Wal|1.0.645
JavaScriptSnippets|xab|1.8.0

(4 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
pythoncet0:30885854
2e7ec940:31000449
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
chatpanelt:31048053
dsvsc021:30996838
g316j359:31013175
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
pythonprt:31056678
dwnewjupytercf:31046870
26j00206:31048877

```

</details>

**Comment by laptomp** [comment]
This issue was addressed in #212090

Two solutions that worked for me are:
- disabling `terminal.integrated.shellIntegration.enabled`
- disabling `terminal.integrated.shellIntegration.enabled`

**Comment by rezadrian01** [comment]
thank you for your solution, the terminal is already fixed
